### DISCUSSION TITLE
Add supoort configuration SIP /USRP / TETRA Logic

### DISCUSSION THREAD

**Discussion by sp2ong** [FM-POLAND/hs_dashboard_pi#31] [Ideas]
Maybe in the future we could add a configuration option in svxlink.conf for LOGICS like 

SIP / USRP / TETRA with svxlink version by Adi DL1HRC

https://github.com/dl1hrc/svxlink

73 Waldek

**Comment by sp2ong** [comment]
If exist in svxlink.conf [UsrpLogic] we can read / display parameters 

**USRP_HOST = 127.0.0.1
USRP_TX_PORT = 44442 
USRP_RX_PORT = 4 4441 
CALL = DB0ABC  
DMRID = 262123**
RPTID= 
DEFAULT_CC = 1  
DEFAULT_TS = 2  
DEFAULT_TG = 2629
**PREAMP = 0
NET_PREAMP = 0**
JITTER_BUFFER_DELAY = 100

**Reply by zochowsd** [reply]
IMHO - if section exist logic should be replaced by some MODE selector condition..... 1. user needs to create the section -> ergo non DH interaction is needed - let's make life easer for user an make it  fully DH oriented. Maybe we should have toogle section with enablers list - simple - section headers - but meybe better is to introduce the mode - like FM HOTSPOT / FM ANYRADIO (no RF config) / TETRA ANYRADIO (no RF)... ect.  Headers of secction  seams to be natural evolution step but will still need a information / documentation for rookie users.  Mode needs maturity level of desiners and good understanding of end-users use cases - imho MODE is better because adress 80%... and it is hard to code / automatize corener cases... and every custom / advanced technice can be suported by linux / terminal / file interation - what is typical behaviur of power users.  Conclusion: --> mode for DH; --> avanced / custom uscases - files / terminal tools.  vy 73 de SP0DZ - cheers

**Reply by zochowsd** [reply]
anoter idea - maybe we need to add [device] tab with all needed parameters to enable a specific configuration / tabs.

**Comment by dl1hrc** [comment]
Hi all,
thank you for starting this discussion, I could imagine that we can implement some interesting functionality into this dashboard for sip/tetra/usrp. I'm not so familiar with php programming but I can certainly do my part.

Sip:
- a phone pad to phone a station
- signalling incoming calls
- something like a small phone book to store vip contacts
- fields for configuration
- control pty: SIP_CTRL_PTY=/tmp/sip_ctrl

Parameters:
USERNAME=2621234    <- I'm normally use the DMR-Id
PASSWORD=*****
SIPPORT=6099
SIPSERVER=svxlink.ham-radio-op.net:6099
SIPREGISTRAR=svxlink.ham-radio-op.net:6099
PHONENUMBERS_TO_TG=262:262,0049:262
#AUTOCONNECT=621
AUTOANSWER=1

Tetra:
- indicate and send SDS's
- showing position reports
- sending/showing state messages
- showing Dapnet messages
- sending Dapnet messages
- control pty: DTMF_CTRL_PTY=/tmp/tetra_dtmf_ctrl
- state pty: STATE_PTY=/tmp/tetra_svxlink_state

For Tetra a bit more parameters are needed for configuration:
BAUD=115200
PORT=/dev/ttyUSB1
ISSI=9999
GSSI=1
MNC=16383
MCC=901
SHARE_USERINFO=1
PROXIMITY_WARNING=3.1
TIME_BETWEEN_SDS=3600
DAPNET_CALLSIGN=xxx
DAPNET_KEY=****
DAPNET_USERNAME=xxxx
DAPNET_PASSWORD=****

and dapnet-Rubrics, ric2issi and so on. But I feel we should start with a lower number of parameters. Since the Sip part will move to contrib directory next time the SipLogic gets it's own configuration file in /etc/svxlink/svxlink.d/SipLogic.conf as well as the TetraLogic will get.

Mni 73s de Adi / DL1HRC
